### DISCUSSION TITLE
Understanding Sampler in .compile method while using gemma models

### DISCUSSION THREAD

**Discussion by Bavalpreet** [keras-team/keras-hub#1468] [Q&A]
sampler="top_k", sampler=keras_nlp.samplers.BeamSampler(num_beams=2)) and by default it is "greedy".
Kindly distinguish between them what does they individually mean?

![Screenshot from 2024-02-25 16-11-02](https://github.com/keras-team/keras-nlp/assets/42736937/a4931266-eee0-4f88-8cc0-3c40c3744cd3)
![Screenshot from 2024-02-25 16-10-47](https://github.com/keras-team/keras-nlp/assets/42736937/849d1c0c-ef7e-458f-b94e-108f42b65978)

**Comment by SinghJagpreet096** [comment]
+1
